resdk/resources/sample.py

```diff
@@ -19,7 +19,10 @@ def get_reads(self):
 
     def get_bam(self):
         """Return ``bam`` object on the sample."""
-        return self.data.get(type='data:alignment:bam:')
+        try:
+            return self.data.get(type='data:alignment:bam:primary:')
```

Comment:
I don't like that `get_bam` function returns primary bam. You should add `get_primary_bam` function to make it cleaner and it can returns normal bam if primary is missing (you can also add flag to enable/disable this behaviour).

Reply:
What about just adding an optional `primary` flag/input to `get_bam` function? Something like:
```def get_bam(self, primary=False):
       if primary:
           try:
               return self.data.get(type='data:alignment:bam:primary:')
           ...
       else:
           return self.data.get(type='data:alignment:bam:')```